inference-engine/src/cldnn_engine/cldnn_infer_request.cpp

```diff
@@ -601,53 +585,32 @@ void CLDNNInferRequest::SetBlob(const std::string& name, const Blob::Ptr &data)
 void CLDNNInferRequest::AllocateInputs() {
     OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "CLDNNInferRequest::AllocateInputs");
     auto inputLayouts = m_graph->GetInputLayouts();
-    // allocate inputs
+
     for (auto& ni : _networkInputs) {
         std::string name = ni.first;
         const TensorDesc& desc = ni.second->getTensorDesc();
 
         if (ColorFormat::NV12 == ni.second->getPreProcess().getColorFormat() &&
             m_graph->getConfig().nv12_two_inputs) {
-            std::vector<Blob::Ptr> blobs;
-            for (auto i = 0; i < desc.getDims()[0]; i++) {
-                cldnn::primitive_id YName(name + "_Y" + std::to_string(i));
-                cldnn::primitive_id UVName(name + "_UV" + std::to_string(i));
-
-                if (inputLayouts.find(YName) == inputLayouts.end()) {
-                    IE_THROW(ParameterMismatch) << "Input layout for " << YName << " is not found";
-                }
-                if (inputLayouts.find(UVName) == inputLayouts.end()) {
-                    IE_THROW(ParameterMismatch) << "Input layout for " << YName << " is not found";
-                }
-                input_alloc(YName, inputLayouts.at(YName));
-                input_alloc(UVName, inputLayouts.at(UVName));
-
-                size_t height = desc.getDims()[2], width = desc.getDims()[3];
-                cldnn::pointer<uint8_t> input_mem_ptr_Y = inputsMemory.at(YName).pointer<uint8_t>();
-                TensorDesc ydesc(Precision::U8, { 1, 1, height, width }, Layout::NHWC);
-                auto blobY = createInputBlob(ydesc, input_mem_ptr_Y.data());
-
-                cldnn::pointer<uint8_t> input_mem_ptr_UV = inputsMemory.at(UVName).pointer<uint8_t>();
-                TensorDesc uvdesc(Precision::U8, { 1, 2, height / 2, width / 2 }, Layout::NHWC);
-                auto blobUV = createInputBlob(uvdesc, input_mem_ptr_UV.data());
-
-                blobs.push_back(make_shared_blob<NV12Blob>(blobY, blobUV));
-            }
-            _inputs[name] = desc.getDims()[0] == 1 ? blobs[0] : make_shared_blob<BatchedBlob>(blobs);
-
+            // don't waste memory for this input, expect SetBlob with decoded input frame
```

Comment:
What if SetBlob won't be called and user will want to get allocated memory from request?

Reply:
Need to add Image2D creation here...